contentreview-impl/impl/src/java/org/sakaiproject/contentreview/impl/turnitin/TurnitinReviewServiceImpl.java

```diff
@@ -906,6 +926,156 @@ public void addTurnitinInstructor(Map userparams) throws SubmissionException, Tr
+			ltiProps.put("custom_allow_non_or_submissions", extraAsnnOpts.get("allow_any_file").toString());
+					//change assignment texts
+			//ONLY FOR TII UK
+			//ltiProps.setProperty("custom_anonymous_marking_enabled", extraAsnnOpts.get("s_paper_check"));
+ 
+			if (extraAsnnOpts.containsKey("exclude_type") && extraAsnnOpts.containsKey("exclude_value")){
+				//exclude type 0=none, 1=words, 2=percentages
+				String typeAux = "words";
+				if(extraAsnnOpts.get("exclude_type").toString().equals("2"))
+					typeAux = "percentage";
+				ltiProps.put("custom_exclude_type", typeAux);
+				ltiProps.put("custom_exclude_value", extraAsnnOpts.get("exclude_value").toString());
+			}
+
+	        ltiProps.put("custom_late_accept_flag", extraAsnnOpts.get("late_accept_flag").toString());
+	        ltiProps.put("custom_report_gen_speed", extraAsnnOpts.get("report_gen_speed").toString());
+	        ltiProps.put("custom_s_view_reports", extraAsnnOpts.get("s_view_report").toString());			
+	        ltiProps.put("custom_submit_papers_to", extraAsnnOpts.get("submit_papers_to").toString());
+
+			if (extraAsnnOpts.containsKey("exclude_biblio")){
+				ltiProps.put("custom_use_biblio_exclusion", extraAsnnOpts.get("exclude_biblio").toString());
+			}
+			if (extraAsnnOpts.containsKey("exclude_quoted")){
+				ltiProps.put("custom_use_quoted_exclusion", extraAsnnOpts.get("exclude_quoted").toString());
+			}
+			
+			//adding callback url
+			String callbackUrl = serverConfigurationService.getString("serverUrl") + "/sakai-contentreview-tool-tii/tii-servlet";
+			log.debug("callbackUrl: " + callbackUrl);
+			ltiProps.put("ext_resource_tool_placement_url", serverConfigurationService.getString("serverUrl") + "/sakai-contentreview-tool-tii/tii-servlet");
+			
+			boolean result = tiiUtil.makeLTIcall(tiiUtil.BASIC_ASSIGNMENT, null, ltiProps);
+			if(!result){
+				log.error("Error making LTI call");//TODO
+				return;
+			}
+			
+			//don't create lti tool if exists
+			if(tiiId == null || ltiId == null){
+				Properties sakaiProps = new Properties();
+				String globalId = tiiUtil.getGlobalTurnitinLTIToolId();
+				sakaiProps.setProperty(LTIService.LTI_TOOL_ID,globalId);
+				
+				String custom = BasicLTIConstants.RESOURCE_LINK_ID + "=" + taskId;
+				custom += ";" + BasicLTIConstants.RESOURCE_LINK_TITLE + "=" + extraAsnnOpts.get("title").toString();
+				log.debug("Storing custom params: " + custom);
+				sakaiProps.setProperty(LTIService.LTI_CUSTOM,custom);
+				sakaiProps.setProperty(LTIService.LTI_SITE_ID,siteId);
+				
+				SecurityAdvisor advisor = new SimpleSecurityAdvisor(sessionManager.getCurrentSessionUserId(), "site.upd", "/site/!admin");
+				Object ltiContent = null;
+				try{
+					securityService.pushAdvisor(advisor);
+					ltiContent = tiiUtil.insertTIIToolContent(globalId, sakaiProps);
+				} catch(Exception e){
+					log.error("createAssignment - insertTIIToolContent: Could not push the security advisor.");
```

Comment:
The log message is wrong, as it could well be the insertTIIToolContent that fails.